File: code_samples/section1/lesson/section1.java
public class section1 {
    // Print each element of an array
    public static void printArrayElements(int[] arr) {
        for (int x : arr) {
            System.out.println(x);
        }
    }

    // Print all pairs of elements of an array
    public static void printElementPairs(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            for (int j = i; j < arr.length; j++) {
                System.out.println(arr[i] + ", " + arr[j]);
            }
        }
    }

    // Print the first and last elements of an array
    public static void printFirstAndLast(int[] arr) {
        if (arr.length > 0) {
            System.out.println(arr[0] + " " + arr[arr.length - 1]);
        }
    }

    // Print each element of an array twice
    public static void printArrayTwice(int[] arr) {
        for (int x : arr) {
            System.out.println(x);
        }
        for (int x : arr) {
            System.out.println(x);
        }
    }

    // Multiply each element of an array by 2
    public static int[] doubleArrayElements(int[] arr) {
        int[] res = new int[arr.length];
        for (int i = 0; i < arr.length; i++) {
            res[i] = arr[i] * 2;
        }
        return res;
    }

    public static void main(String[] args) {
        int[] arr = { 1, 2, 3, 4, 5 };

        System.out.println("Ex 1:");
        printArrayElements(arr);

        System.out.println("\nEx 2:");
        printElementPairs(arr);

        System.out.println("\nEx 3:");
        printFirstAndLast(arr);

        System.out.println("\nEx 4:");
        printArrayTwice(arr);

        System.out.println("\nEx 5:");
        int[] res = doubleArrayElements(arr);
        for (int x : res) {
            System.out.println(x);
        }
    }
}

File: code_samples/section1/lesson/section1.py
"""Section 1 - Foundations"""

#pylint: disable=consider-using-enumerate

def print_array_elements(arr):
    """Prints all elements of an array"""
    for x in arr:
        print(x)

def print_element_pairs(arr):
    """Prints all pairs of elements of an array"""
    for i in range(len(arr)):
        for j in range(i, len(arr)):
            print(arr[i], arr[j])

def print_first_and_last(arr):
    """Prints the first and last elements of the list"""
    if arr:
        print(arr[0], arr[-1])

def print_array_twice(arr):
    """Prints the list twice"""
    for x in arr:
        print(x)
    for x in arr:
        print(x)

def double_array_elements(arr):
    """Returns a list of x*2 for each x in arr"""
    return [x * 2 for x in arr]

lst = [1, 2, 3, 4, 5]

print("Ex 1:")
print_array_elements(lst)

print("\nEx 2:")
print_element_pairs(lst)

print("\nEx 3:")
print_first_and_last(lst)

print("\nEx 4:")
print_array_twice(lst)

print("\nEx 5:")
res = double_array_elements(lst)
for n in res:
    print(n)
